### PR TITLE
Support tighter overrides for Contextual Menu Items

### PR DESCRIPTION
**Pull request by ThomasMichon**
## Previous Behavior

Right now, command bar and context menu items only support a shared `onRender` override, which does not supply a `defaultRender` callback to use to handle nuance in context. `ICommandBarItemProps` supports a `commandBarButtonAs`, but this only works in the main command bar and not the overflow menu or normal context menus. The `contextualMenuItemAs` is only available for the `ContextualMenu` as a whole and cannot be customized per-item. It is also not composable with its default render behavior.

## New Behavior

Updated consumption of `contextualMenuItemAs` so that behaviors compose with a per-item behavior and with the default value. Updated the interface to use `IComponentAs`, which is backwards-compatible with the existing behavior. (The whole behavior is backwards-compatible, because existing implementors aren't expected to be using `defaultRender` anyway).

Unfortunately, since `contextualMenuItemAs` does not support overriding the `button` wrapper, added a new `contextualMenuItemWrapperAs` field which can be used to include the `button` or other render layer.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/24d5ad1480e678e6201f99875068c8cefee2b034'>24d5ad1480e678e6201f99875068c8cefee2b034</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/BaseButton-mount_base.html">806</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/BaseButton-mount.html">792</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Breadcrumb-mount_base.html">2053</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Breadcrumb-mount.html">2081</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Checkbox-mount_base.html">2087</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Checkbox-mount.html">2133</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/CheckboxBase-mount_base.html">1836</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/CheckboxBase-mount.html">1873</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ChoiceGroup-mount_base.html">3495</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ChoiceGroup-mount.html">3543</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ComboBox-mount_base.html">794</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ComboBox-mount.html">806</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/CommandBar-mount_base.html">7726</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/CommandBar-mount.html">7770</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ContextualMenu-mount_base.html">13310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ContextualMenu-mount.html">14096</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DefaultButton-mount_base.html">944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DefaultButton-mount.html">954</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRow-mount_base.html">2668</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRow-mount.html">2639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRowFast-mount_base.html">2612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRowFast-mount.html">2647</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRowNoStyles-mount_base.html">2429</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DetailsRowNoStyles-mount.html">2446</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Dialog-mount_base.html">3251</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Dialog-mount.html">3228</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DocumentCardTitle-mount_base.html">301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/DocumentCardTitle-mount.html">296</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Dropdown-mount_base.html">2473</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Dropdown-mount.html">2468</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/FocusTrapZone-mount_base.html">1420</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/FocusTrapZone-mount.html">1424</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/FocusZone-mount_base.html">1387</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/FocusZone-mount.html">1385</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-mount_base.html">45355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-mount.html">45254</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-virtual-rerender_base.html">21969</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-virtual-rerender.html">21987</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">62592</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedList-virtual-rerender-with-unmount.html">62954</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-mount_base.html">296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-mount.html">304</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-virtual-rerender_base.html">279</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-virtual-rerender.html">274</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">292</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/IconButton-mount_base.html">1325</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/IconButton-mount.html">1333</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Label-mount_base.html">459</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Label-mount.html">454</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Layer-mount_base.html">3455</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Layer-mount.html">3413</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Link-mount_base.html">526</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Link-mount.html">519</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/MenuButton-mount_base.html">1161</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/MenuButton-mount.html">1159</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/MessageBar-mount_base.html">25976</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/MessageBar-mount.html">26531</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Nav-mount_base.html">2319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Nav-mount.html">2372</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/OverflowSet-mount_base.html">1041</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/OverflowSet-mount.html">995</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Panel-mount_base.html">2178</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Panel-mount.html">2185</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Persona-mount_base.html">933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Persona-mount.html">928</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Pivot-mount_base.html">1070</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Pivot-mount.html">1088</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/PrimaryButton-mount_base.html">1052</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/PrimaryButton-mount.html">1054</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Rating-mount_base.html">5716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Rating-mount.html">5738</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SearchBox-mount_base.html">1134</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SearchBox-mount.html">1105</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Shimmer-mount_base.html">2339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Shimmer-mount.html">2370</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Slider-mount_base.html">1686</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Slider-mount.html">1672</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SpinButton-mount_base.html">3450</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SpinButton-mount.html">3436</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Spinner-mount_base.html">530</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Spinner-mount.html">538</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SplitButton-mount_base.html">2173</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SplitButton-mount.html">2191</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Stack-mount_base.html">543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Stack-mount.html">537</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/StackWithIntrinsicChildren-mount_base.html">1110</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/StackWithIntrinsicChildren-mount.html">1098</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/StackWithTextChildren-mount_base.html">3375</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/StackWithTextChildren-mount.html">3371</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SwatchColorPicker-mount_base.html">7365</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/SwatchColorPicker-mount.html">7354</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/TagPicker-mount_base.html">1718</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/TagPicker-mount.html">1733</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Text-mount_base.html">496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Text-mount.html">506</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/TextField-mount_base.html">1151</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/TextField-mount.html">1137</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-mount_base.html">1023</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-mount.html">1034</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-virtual-rerender_base.html">810</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-virtual-rerender.html">803</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1557</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Toggle-mount_base.html">800</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/Toggle-mount.html">793</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/buttonNative-mount_base.html">259</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27907/perf-test/buttonNative-mount.html">259</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Review by GeoffCoxMSFT — APPROVED** [review]

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx [hunk 1]
```diff
@@ -1122,7 +1150,16 @@ export const ContextualMenuBase: React.FunctionComponent<IContextualMenuProps> =
...
-      const { contextualMenuItemAs: ChildrenRenderer = ContextualMenuItem } = props;
+      let ChildrenRenderer: IComponentAs<IContextualMenuItemProps> = ContextualMenuItem;
+
+      if (item.contextualMenuItemAs) {
+        ChildrenRenderer = composeComponentAs(item.contextualMenuItemAs, ChildrenRenderer);
+      }
+
+      if (props.contextualMenuItemAs) {
+        ChildrenRenderer = composeComponentAs(props.contextualMenuItemAs, ChildrenRenderer);
+      }
```
Comment (smhigley) [inline packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:1161]: Should `props.contextualMenuItemAs` override `item.contextualMenuItemAs`? I think I'd lean towards the `item` prop taking precedence since it's more specific, but let me know if there's another reason to keep this order

**Review by smhigley — COMMENTED** [review]
I'm still a little worried that the `IComponentAs` will imply we support a `defaultRender` prop on the `contextualMenuItemAs`/ `focusZoneAs` default components, but it looks like this isn't the only place we have that pattern, so I don't think it should hold up this change.

Other than that, just the one question about precedence for `props.contextualMenuItemAs` vs. `item.contextualMenuItemAs`

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChart.test.tsx.snap [hunk 2]
```diff
@@ -101,7 +101,7 @@ exports[`HeatMapChart - mouse events Should render callout correctly on mouseove
         transform="translate(40, 0)"
       />
       <g
-        aria-label="Mar/04, p2. Execllent (0-200), 25/2479. Due to unexpected heavy rain, all the pollutants are washed
+        aria-label="Mar/03, p2. Execllent (0-200), 25/2479. Due to unexpected heavy rain, all the pollutants are washed
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChart.test.tsx.snap:104]: Updates in this file don't seem related to the change. Could you confirm.

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx (hunk 1 above)
Comment (ThomasMichon) [reply packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:1161]: The way this was done in `CommandBar` was that the whole-bar override supersedes the `ContextualMenu` one. But I agree it would make more sense to flip it, or at least for there also to be the ability for there to be an override-able "base" component. This is worth a follow-up for that capability.

File: packages/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChart.test.tsx.snap (hunk 2 above)
Comment (ThomasMichon) [reply packages/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChart.test.tsx.snap]: Reverted. This is some artifact of running the unit tests locally.

**Review by smhigley — APPROVED** [review]
